src/compiler/crystal/tools/playground/public/session.js

```diff
@@ -297,7 +297,8 @@ Playground.Session = function(options) {
   this.editor._playgroundSession = this;
 
   this.connect = function() {
-    this.ws = new WebSocket("ws://" + location.host + "/client");
+    var socketProtocol = location.protocol === "https:" ? "wss:" : "ws:";
+    this.ws = new WebSocket(socketProtocol + "//" + location.host + "/client");
```

Comment:
Have we tried just using "//host/client" with no scheme? At least on HTTP that ends up using the current protocol.

Reply:
I don't think we should use protocol-relative URLs. It's considered an antipattern and shouldn't be advocated. This is easy to solve with JavaScript, let's just do it that way.

Reply:
OK

Reply:
I agree with @straight-shoota in the anti-pattern case.  Nonetheless for the record, neither [MDN's WebSocket][0] nor the [WebSocket spec][1] specify a scheme state override passed to the URL parser which seems to result in a [no scheme state][2] -> [relative state][3] where the base URL's scheme will be used.  If I interpret that correctly, that suggests the fallback will be **HTTP** or **HTTPS** in this case which is not exactly what we want even though the default ports for those are the same as **WS** and **WSS** respectively.

[0]: https://developer.mozilla.org/en-US/docs/Web/API/WebSocket
[1]: https://html.spec.whatwg.org/multipage/web-sockets.html
[2]: https://url.spec.whatwg.org/#no-scheme-state
[3]: https://url.spec.whatwg.org/#relative-state